src/rgw/rgw_bucket.cc

```diff
@@ -532,9 +533,15 @@ int rgw_remove_bucket(RGWRados *store, rgw_bucket& bucket, bool delete_children)
   if (ret < 0)
     return ret;
 
+  string prefix, delimiter;
+
+  ret = abort_bucket_multiparts(store, cct, info, prefix, delimiter);
+  if (ret < 0) {
+    return ret;
+  }
+
```

Comment:
why did this call need to move up? i think we want it to come after the `list_objects/rgw_remove_object` loop in case we pass `delete_children=false`, so we can return `-ENOTEMPTY` without mutating the bucket